code/FulpstationCode/blueshift_gauntlets/bsg.dm

```diff
@@ -0,0 +1,80 @@
+var/isPhased = FALSE
+var/isDrapeSpawned = FALSE
+var/isCauterySpawned = FALSE
+var/bsgtrait = TRAIT_BLUESPACE_SHIFTING
+
+//Action Buttons
+var/datum/action/set_phase/setphase = new/datum/action/set_phase()
+var/datum/action/summon_implements/summonimplement = new/datum/action/summon_implements()
+
+/datum/action/set_phase //Create the button for toggling the gloves phase.
+	name = "Change Phase - Enable or disable the gloves phasing device."
+	icon_icon = 'icons/mob/actions/actions_items.dmi' //placeholder
+	button_icon_state = "neckchop" //placeholder
+
+/datum/action/set_phase/Trigger() //Toggle the gloves phase between on or off.
+	if(owner.incapacitated())
+		to_chat(owner, "<span class='warning'>You can't use [name] while you're incapacitated!</span>")
+		return
+	if(isPhased == FALSE)
+		to_chat(owner, "<span class ='notice'>You activate the gauntlets, preparing them to phase!</span>")
+		isPhased = TRUE
+		owner.visible_message("[owner.name] activates their Blueshift Gauntlets with a low hum.")
+	else
+		to_chat(owner, "<span class ='notice'>You deactivate the gauntlets, removing them from the bluespace fold!</span>")
+		isPhased = FALSE
+		owner.visible_message("[owner.name] deactivates their Blueshift Gauntlets with a soft beep.")
+		playsound(owner,'sound/machines/beep.ogg',1,TRUE)
+
+/datum/action/summon_implements
+	name = "Summon Implements - Summon a holographic cautery or set of drapes for starting operations."
+	icon_icon = 'icons/mob/actions/actions_items.dmi' //placeholder
+	button_icon_state = "neckchop" //placeholder
+
+/datum/action/summon_implements/Trigger()
+	if(owner.incapacitated())
+		to_chat(owner, "<span class='warning'>You can't use [name] while you're incapacitated!</span>")
+		return
+	if(isPhased == FALSE)
+		to_chat(owner, "<span class ='warning'>You can't use [name] while the gauntlets arent phased!</span>")
+		owner.visible_message("[owner.name] tries to summon a tool while the gauntlets are off with a dull beep.")
+		playsound(owner,'sound/machines/buzz-two.ogg',1,TRUE)
+	else
+		to_chat(owner, "<span class ='notice'>You deactivate the gauntlets, removing them from the bluespace fold!</span>")
```

Comment:
?

Reply:
huh?

Reply:
the gauntlets turn themselves off if you take them off while they're on is what that does essentially, prevents the var getting stuck turned on after being removed or other werid scenarios like that by resetting the var